src/config.rs

```diff
@@ -1059,6 +1073,132 @@ impl RaftDbConfig {
     }
 }
 
+#[derive(Clone, Copy, Debug)]
+pub enum DBType {
+    Kv,
+    Raft,
+}
+
+pub struct DBConfigManger {
+    db: Arc<DB>,
+    db_type: DBType,
+}
+
+impl DBConfigManger {
+    pub fn new(db: Arc<DB>, db_type: DBType) -> Self {
+        DBConfigManger { db, db_type }
+    }
+}
+
+impl DBConfigManger {
+    fn set_db_config(&self, opts: &[(&str, &str)]) -> Result<(), Box<dyn Error>> {
+        self.db.set_db_options(opts)?;
+        Ok(())
+    }
+
+    fn set_cf_config(&self, cf: &str, opts: &[(&str, &str)]) -> Result<(), Box<dyn Error>> {
+        self.validate_cf(cf)?;
+        let handle = get_cf_handle(&self.db, cf)?;
+        self.db.set_options_cf(handle, opts)?;
+        // Write config to metric
+        for (cfg_name, cfg_value) in opts {
+            let cfg_value = match cfg_value {
+                v if *v == "ture" => Ok(1f64),
+                v if *v == "false" => Ok(0f64),
+                v => v.parse::<f64>(),
+            };
+            if let Ok(v) = cfg_value {
+                CONFIG_ROCKSDB_GAUGE
+                    .with_label_values(&[cf, cfg_name])
+                    .set(v);
+            }
+        }
+        Ok(())
+    }
+
+    fn set_block_cache_size(&self, cf: &str, size: ReadableSize) -> Result<(), Box<dyn Error>> {
+        self.validate_cf(cf)?;
+        let handle = get_cf_handle(&self.db, cf)?;
+        let opt = self.db.get_options_cf(handle);
+        opt.set_block_cache_capacity(size.0)?;
+        // Write config to metric
+        CONFIG_ROCKSDB_GAUGE
+            .with_label_values(&[cf, "block_cache_size"])
+            .set(size.0 as f64);
+        Ok(())
+    }
+
+    fn validate_cf(&self, cf: &str) -> Result<(), Box<dyn Error>> {
+        match (self.db_type, cf) {
+            (DBType::Kv, CF_DEFAULT)
+            | (DBType::Kv, CF_WRITE)
+            | (DBType::Kv, CF_LOCK)
+            | (DBType::Kv, CF_RAFT)
+            | (DBType::Raft, CF_DEFAULT) => Ok(()),
+            _ => Err(format!("invalid cf {:?} for db {:?}", cf, self.db_type).into()),
+        }
+    }
+}
+
+impl ConfigManager for DBConfigManger {
+    fn dispatch(&mut self, change: ConfigChange) -> Result<(), Box<dyn Error>> {
+        let change_str = format!("{:?}", change);
+        let mut change: Vec<(String, ConfigValue)> = change.into_iter().collect();
+        let cf_config = change.drain_filter(|(name, _)| name.ends_with("cf"));
```

Comment:
what would the change string be like?

Reply:
It will be the field name in the config struct like `defaultcf`.

Reply:
The `change_str ` will be something like `{"rocksdb": {"defaultcf": {"target_file_size_base": 67108864b, "block_cache_size": 8388608b}, "max_background_jobs": 2}}`.